### ISSUE TITLE
License: GPL-3.0-only or GPL-3.0-or-later?

### ISSUE THREAD

**Issue by fkohrt** [NMarkgraf/typography.py#7]
Could you provide further details on the license you chose, i.e. if it's "GPL-3.0-only" or "GPL-3.0-or-later"? See the article "[For Clarity's Sake, Please Don't Say “Licensed under GNU GPL 2”!](https://www.gnu.org/licenses/identify-licenses-clearly.html)" for further information.

E.g. to visibly choose GPL-3.0-or-later according to the [REUSE specification](https://reuse.software/), you could apply the following patch:

```patch
--- /dev/fd/63	2021-07-11 15:15:26.401000000 +0200
+++ typography.py	2021-07-11 15:15:01.081000000 +0200
@@ -1,6 +1,10 @@
 #!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 
+# SPDX-FileCopyrightText: 2017-2021 by Norman Markgraf <nmarkgraf@hotmail.com>
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
 """
   Quick-Pandoc-Typographie-Filter: typography.py
 
```

**Comment by NMarkgraf** [comment]
Hi Florian,

Thanks a lot for the idea. I will think about it, but please, why should I support SPDX? - I don't know much about it, but I am curious. What is it, who use it and why should this project use/support it too?

Yours
 Norman

**Comment by fkohrt** [comment]
Hi Norman,

thanks for ea8886c! To be honest, I missed the [license text below](https://github.com/NMarkgraf/typography.py/blob/3fa78db/typography.py#L49-L62), my bad.

> What is it

SPDX is an attempt to standardize license identifiers. Other projects can refer to SPDX identifiers and thus improve the machine readability of their content's license. E.g. npm package descriptions contain a [license field](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#license) that must be an SPDX identifier. The REUSE specification is another project that makes use of them. It suggests how individual software projects (independent of language) describe their source code's licenses in a machine readable way. It partly builds on the [`debian/copyright` file specification](https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/) and allows to track the license status down to individual source code files.

> who use it

I am not sure about the adoption of REUSE, I was merely suggesting the only (formalized) standard I knew of to communicate licenses. I could have asked to just write the license choice anywhere (given that I did not notice that you did that already), but I figured I'd recommend a standardized way to do this.

> why should this project use/support it too

(1) Others are more likely to correctly identify your license choice. (2) Computers are more likely to identify your license choice. (3) Integration into other projects that conform to REUSE becomes easier.

Best,
Florian
